src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -261,15 +262,25 @@ public void write() {
               }
             } else {
               SinkRecord projectedRecord = SchemaUtils.project(record, currentSchema, compatibility);
-              writeRecord(projectedRecord);
-              buffer.poll();
-              if (shouldRotate(now)) {
+              String partition = partitioner.encodePartition(projectedRecord);
+              if (currentPartition != null && !currentPartition.equals(partition)) {
+                currentPartition = partition;
```

Comment:
Thanks for reply! You are right, this will no work with `FieldPartitioner` and it assumes that timestamps will be in order.

The idea here is to provide data for external ETL pipeline in predictable time even if number of messages in a chunk is not reached. Do you have any idea how can I achieve this with kafka connect right now?

Thanks!